### PR TITLE
test: enhance error accumulator and form builder tests, add marshaller tests

### PR DESCRIPTION
**Pull request by Whitea029**
**Describe the change**
enhance error accumulator and form builder tests, add marshaller tests

**Describe your solution**
added some edge case tests to improve test coverage

Issue: #997

### CODE REVIEW

File: internal/form_builder.go [hunk 1]
```diff
@@ -53,6 +53,9 @@ func (fb *DefaultFormBuilder) createFormFile(fieldname string, r io.Reader, file
 }
 
 func (fb *DefaultFormBuilder) WriteField(fieldname, value string) error {
+	if fieldname == "" {
+		return fmt.Errorf("fieldname cannot be empty")
```
[the remark this answers is not in the document]
Comment (Whitea029) [reply internal/form_builder.go:57]: Personally, I don't think there is much need for this change

[the remark this answers is not in the document]
Comment (kmesiab) [reply internal/form_builder.go:57]: The project is pretty clear about its opinions on errors:

 https://github.com/search?q=repo%3Asashabaranov%2Fgo-openai+errors.New&type=code

**Comment by Whitea029** [comment]
@sashabaranov Merge conflicts have been resolved. Let me know if there’s anything else needed before merging!

**Review by sashabaranov — APPROVED** [review]
